src/js/components/List/List.js

```diff
@@ -24,6 +24,13 @@ const StyledList = styled.ul`
 
 const StyledItem = styled(Box)`
   ${props => props.onClick && `cursor: pointer;`}
+  &:hover {
+    ${props =>
+      props.theme.list.hover &&
+      props.theme.list.hover.item &&
+      props.theme.list.hover.item.background &&
+      backgroundStyle(props.theme.list.hover.item.background, props.theme)}
```

Comment:
I think this implies that any List will have per-item hover styling, whether or not that item is interactive in any way. Something seems a little off here. Perhaps I am missing something?

Reply:
The intention of this is to allow a list to have per-item hover styling, specifically when it doesn't have `onClick`. The motivation for this was because of HPE design system designs that called for a list item hover state regardless of if it has an onClick or not.

Some list items may have an action button, in which case the entire list item will not have an `onClick`. However, having the the hover behavior helps with the visual experience of differentiating list items. If we don't feel like this is a behavior we want to support though, we would want to revisit the List designs to modify what is currently noted as hover behavior.

Reply:
Will this work on Menu items?

Reply:
Hey, @cliffhall! This will specifically be affecting the List component. Menu is not built with List, so this change will not be affecting Menu. 

If you're looking for styling customization on hover for Menu that you're unable to achieve, feel free to file a new issue!